src/librbd/ImageWatcher.cc

```diff
@@ -844,7 +844,6 @@ void ImageWatcher::handle_payload(const FlattenPayload &payload,
 
 void ImageWatcher::handle_payload(const ResizePayload &payload,
 				  bufferlist *out) {
-  RWLock::RLocker l(m_image_ctx.owner_lock);
```

Comment:
I think it would be ok since the resize state machine can handle the lost exclusive lock case.